velox/functions/prestosql/ArrayFunctions.h

```diff
@@ -610,6 +610,78 @@ struct ArrayFrequencyFunction {
   folly::F14FastMap<arg_type<T>, int> frequencyCount_;
 };
 
+// Array<Varchar> to handle nested arrays
+template <typename TExecParams>
+struct ArrayFrequencyFunction<TExecParams, Array<Varchar>> {
+  VELOX_DEFINE_FUNCTION_TYPES(TExecParams);
+
+  // Helper struct to hash Array<Varchar>
+  struct ArrayVarcharKey {
+    std::vector<std::optional<std::string>> elements;
+
+    ArrayVarcharKey() = default;
+
+    explicit ArrayVarcharKey(const arg_type<Array<Varchar>>& array) {
+      for (const auto& item : array) {
+        if (item.has_value()) {
+          elements.emplace_back(item.value().getString());
+        } else {
+          elements.emplace_back(std::nullopt);
+        }
+      }
+    }
+
+    bool operator==(const ArrayVarcharKey& other) const {
+      return elements == other.elements;
+    }
+  };
+
+  struct ArrayVarcharKeyHasher {
+    std::size_t operator()(const ArrayVarcharKey& key) const {
+      std::size_t hash = 0;
+      for (const auto& elem : key.elements) {
+        if (elem.has_value()) {
+          hash = folly::hash::hash_combine(
+              hash, std::hash<std::string>{}(elem.value()));
```

Comment:
Why cant we store the hash when we read an item from the array and store that along with a reference to item and a frequency count. Keep updating the frequency count on a collision or adding a new entry when no collision, this will be less expensive than maintaining a copy via ArrayVarcharKey.